src/analysis/processing/qgsalgorithmrasterstackposition.cpp

```diff
@@ -0,0 +1,404 @@
+{
+  return QObject::tr( "lowestpositioninrasterstack" );
+}
+
+QStringList QgsRasterStackLowestPositionAlgorithm::tags() const
+{
+  return QObject::tr( "cell,lowest,position,pixel,stack" ).split( ',' );
+}
+
+QString QgsRasterStackLowestPositionAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "The lowest position algorithm evaluates on a cell-by-cell basis the position "
+                      "of the raster with the highest value in a stack of rasters. Position counts start "
+                      "with 1 and range to the total number input rasters. The order of the input "
+                      "rasters is relevant for the algorithm. If multiple rasters feature the lowest value, "
+                      "the first raster will be used for the position value.\n "
+                      "If multiband rasters are used in the data raster stack, the algorithm will always "
+                      "perform the analysis on the first band of the rasters - use GDAL to use other bands in the analysis. "
+                      "Any NoData cells in the raster layer stack will result in a NoData cell "
+                      "in the output raster unless the ignore NoData parameter is checked. "
+                      "The output NoData value can be set manually. The output rasters extent and resolution "
+                      "is defined by a reference raster layer and is always of int32 type." );
+}
+
+QgsRasterStackLowestPositionAlgorithm *QgsRasterStackLowestPositionAlgorithm::createInstance() const
+{
+  return new QgsRasterStackLowestPositionAlgorithm();
+}
+
+int QgsRasterStackLowestPositionAlgorithm::findPosition( std::vector< std::unique_ptr<QgsRasterBlock> > &inputBlocks, int &row, int &col, bool &noDataInRasterBlockStack )
+{
+  int lowestPosition = 0;
+
+  //auxiliary variables
+  int inputBlocksCount = inputBlocks.size();
+  int currentPosition = 0;
+  int noDataCount = 0;
+  double firstValue = mNoDataValue;
+  bool firstValueIsNoData = true;
+
+  while ( firstValueIsNoData && ( currentPosition < inputBlocksCount ) )
+  {
+    //check if all blocks are nodata/invalid
+    std::unique_ptr<QgsRasterBlock> &firstBlock = inputBlocks.at( currentPosition );
+    firstValue = firstBlock->valueAndNoData( row, col, firstValueIsNoData );
+
+    if ( !firstBlock->isValid() || firstValueIsNoData )
+    {
+      noDataInRasterBlockStack = true;
+      noDataCount++;
+    }
+    else
+    {
+      lowestPosition = currentPosition;
+    }
+    currentPosition++;
+  }
+
+  if ( noDataCount == inputBlocksCount )
+  {
+    noDataInRasterBlockStack = true;
+    return -1; //all blocks are NoData
+  }
+  else
+  {
+    //scan for the lowest value
+    while ( currentPosition < inputBlocksCount )
+    {
+      std::unique_ptr< QgsRasterBlock > &currentBlock = inputBlocks.at( currentPosition );
+
+      bool currentValueIsNoData = false;
+      double currentValue = currentBlock->valueAndNoData( row, col, currentValueIsNoData );
```

Comment:
Would be faster if you stored a list of values/nodatas in the initial iteration -- otherwise in most cases (where no nodata is present) you're fetching each pixel value twice.

Reply:
Actually both iterations run consecutively so that no pixel values are fetched twice/no blocks are visited twice. The second iteration picks up the work as soon as an `inputBlock` does return a value instad of noData. It then searches for the lowest/highest value among the other `inputBlocks`. Or did you mean something else?

Reply:
Ok, I misinterpreted -- all good!